src/hubspot/index.ts

```diff
@@ -0,0 +1,51 @@
+import fetch, { RequestInit } from 'node-fetch';
+import qs from 'qs';
+
+export default class Hubspot {
+  constructor(
+    private readonly apiBaseUrl: string,
+    private readonly hapikey: string,
+  ) {}
+
+  get(resource: string, config?: HubspotRequestConfig) {
+    return this.query(resource);
+  }
+
+  private async query(resource: string, params?: any, init?: RequestInit) {
+    const pagination: any = {};
+    const exec = () =>
+      fetch(
+        `${this.apiBaseUrl}${resource}?` +
+          qs.stringify({
+            hapikey: this.hapikey,
+            ...params,
+            ...pagination,
+          }),
+        init,
+      );
+    let data: any = null;
+    let results: any[] | undefined = undefined;
+    do {
+      const res = await exec();
+      data = await res.json();
+      pagination.after = data?.paging?.next?.after;
+      if (data?.results) {
+        if (!results) {
+          results = [];
+        }
+        results.push(...data?.results);
+      }
+    } while (data?.results && data?.paging?.next?.after);
+    if (data.status === 'error' && data.category === 'INVALID_AUTHENTICATION') {
+      throw new Error('Invalid authentication');
+    }
+    return results || data;
```

Comment:
I've already mentioned this in my other comment, but it'll need to look more similar to: https://github.com/JupiterOne/graph-rapid7/blob/1dc194669abdf5cce8eada59cad1f67c743f0efc/src/client.ts#L71